builtin/providers/aws/resource_aws_kinesis_stream.go

```diff
@@ -95,10 +95,10 @@ func resourceAwsKinesisStreamCreate(d *schema.ResourceData, meta interface{}) er
 			sn, err)
 	}
 
-	s := streamRaw.(kinesisStreamState)
+	s := streamRaw.(*kinesisStreamState)
 	d.SetId(s.arn)
 	d.Set("arn", s.arn)
-	d.Set("shard_count", s.shardCount)
+	d.Set("shard_count", len(s.openShards))
```

Comment:
There is no actual change in behavior here, the value of `s.shardCount` before was the number of open shards. And since I'm now storing the actual list of open shard ids, we can derive the shard count from it rather than storing it separately.